File: backend/Models/Todo.cs
using System.ComponentModel.DataAnnotations;

namespace LETM.Models
{
    public class Todo
    {
        public int Id { get; set; }

        [Required]
        public string Title { get; set; } = string.Empty;

        public string? Description { get; set; }

        public bool IsCompleted { get; set; } = false;

        public DateTimeOffset CreatedAt { get; set; }

        public DateTimeOffset UpdatedAt { get; set; }
    }
}

File: frontend/app/types/todo.ts
export interface Todo {
    id: number;
    title: string;
    description?: string;
    isCompleted: boolean;
    createdAt: string; // We technically don't need these defined here since we aren't using them
    updatedAt: string; // But adding them for consistency with the backend model, plus in a real project they could be useful
}